![header](https://github.com/KGrond/test_workshop/blob/912d63304d4f33b774b487a112cd600944264ed3/assets/images/git_header_v2.png)

# Materials for the 2025 Alaska INBRE Data Science Core Genomics Workshop 

Welcome! This repository contains the materials that we used during our workshop. The easiest way to access all the materials is to [download the whole repository](https://github.com/KGrond/test_workshop).


## [![Lectures](https://img.shields.io/badge/Lectures-blue)](https://github.com/KGrond/test_workshop/tree/a53466f83521559cb00afc813fd891cc46f3d58a/lectures) Lecture Slides

1. [Introduction to Nanopore Sequencing](https://raw.githubusercontent.com/KGrond/test_workshop/main/lectures/DSC-Workshop2025_Mod-1-2_IntroNanopore.pdf)
2. [Cloud Computing and HPCs](https://raw.githubusercontent.com/KGrond/test_workshop/tree/main/lectures/DSC-Workshop2025_Mod-2_Cloud-Computing-and-HPCs.pdf)
3. [Pipelines, Containers, and Workflow Management Systems](https://raw.githubusercontent.com/KGrond/test_workshop/tree/main/lectures/DSC-Workshop2025_Mod-3_pipelines-workflows-containers.pdf)
4. [Principles of Reproducible Research](https://raw.githubusercontent.com/drownlab/![dsc_workshop_2025/tree/main/lectures/DSC-Workshop2025_Mod-3_Reproducible-Research-Principles.pdf)
5. [Version Control with Git](https://raw.githubusercontent.com/KGrond/test_workshop/tree/main/lectures/DSC-Workshop2025_Mod-3_VersionControl.pdf)

## [![Notebooks](https://img.shields.io/badge/Notebooks-blue)](https://github.com/KGrond/test_workshop/tree/a53466f83521559cb00afc813fd891cc46f3d58a/notebooks) Jupyter Notebooks 

### 1. Nanopore Analysis
* [Epi2me setup](https://github.com/KGrond/test_workshop/tree/main/notebooks/nanopore_analysis/DSC_Module_2_Epi2meSetup.ipynb)  
* [Epi2me bacterial-genomes workflow](https://github.com/KGrond/test_workshop/tree/main/notebooks/nanopore_analysis/DSC_Module_2_wf-bacterial-genomes.ipynb)  
* [Epi2me metagenomics workflow](https://github.com/KGrond/test_workshop/tree/main/notebooks/nanopore_analysis/DSC_Module_2_wf-metagenomics.ipynb)  
* [Walkthrough of the get_all_stats.sh script](https://github.com/KGrond/test_workshop/tree/main/notebooks/nanopore_analysis/DSC_Module_4_get_stats.ipynb) 
* [Walkthrough of the install_bioinfo_tools.sh script](https://github.com/KGrond/test_workshop/tree/main/notebooks/nanopore_analysis/DSC_Module_4_install_bioinfo_tools.ipynb)
* [Adaptive sampling analysis pipeline](https://github.com/KGrond/test_workshop/tree/main/notebooks/nanopore_analysis/DSC_Module_4_process_AS.ipynb)


### 2. Version Control with Git
* Lesson 1: [Configure Git](https://github.com/KGrond/test_workshop/tree/main/notebooks/git/Lesson1_Setting_up_git.ipynb)  
* Lesson 2: [Getting started](https://github.com/KGrond/test_workshop/tree/main/notebooks/git/Lesson2_Create_Project_and_View_History_Ignoring_Files.ipynb)  
* Lesson 3: [Connecting to Github](https://github.com/KGrond/test_workshop/tree/main/notebooks/git/Lesson3_Connecting_to_Github.ipynb)  
* Lesson 4: [Branching and merging](https://github.com/KGrond/test_workshop/tree/main/notebooks/git/Lesson4_Branching_merging.ipynb)  
* Lesson 5: [Collaborating with Git](https://github.com/KGrond/test_workshop/tree/main/notebooks/git/Lesson5_Collaborating.ipynb)  
* Lesson 6: [Solve a merge conflicts](https://github.com/KGrond/test_workshop/tree/main/notebooks/git/Lesson6_Managing_Conflict.ipynb)  
* Lesson 7: [RStudio & Git](https://github.com/KGrond/test_workshop/tree/main/notebooks/git/Lesson7_RStudioServer_Git.ipynb)  


## [![Scripts](https://img.shields.io/badge/Scripts-blue)](https://github.com/KGrond/test_workshop/tree/a53466f83521559cb00afc813fd891cc46f3d58a/scripts) Shell and R scripts used in this workshop

* Visualize sequencing data in R through [violin plots](https://github.com/KGrond/test_workshop/tree/main/scripts/violinplot_treatments_isolates.R) 
* Create [bar plots](https://github.com/KGrond/test_workshop/tree/main/scripts/plot_treatments_isolates.R) in R to compare different sequencing metrics.
* [Install bioinformatics tools](https://github.com/KGrond/test_workshop/tree/main/scripts/install_bioinfo_tools.sh) on a Google Cloud Vertex AI instance.
* [Collect SeqKit stats](https://github.com/KGrond/test_workshop/tree/main/scripts/get_all_stats.sh) for various FASTQ outputs from the adaptive sampling analysis pipeline.
* [Process and analyze nanopore adaptive sampling data](https://github.com/KGrond/test_workshop/tree/main/scripts/as_processing_cloud_v2.sh)


## [![Handouts](https://img.shields.io/badge/Handouts-blue)](https://github.com/KGrond/test_workshop/tree/a53466f83521559cb00afc813fd891cc46f3d58a/handouts) Nanopore Handouts: Descriptions of the mock communities and different library prepapration methods 

* [Mock Communities](https://github.com/KGrond/test_workshop/tree/main/handouts/Mock_Communities)
* [Sequencing protocols & Information](https://github.com/KGrond/test_workshop/tree/main/handouts/Protocols)


